tests/templates/test_subroutines/test_qrom.py

```diff
@@ -192,6 +193,14 @@ def test_decomposition(self):
         for op1, op2 in zip(qrom_decomposition, expected_gates):
             qml.assert_equal(op1, op2)
 
+    def test_decomposition_new(self):
+        """Tests the decomposition rule implemented with the new system."""
+        op = qml.QROM(
+            ["1", "0", "0", "1"], control_wires=[0, 1], target_wires=[2], work_wires=[3], clean=True
+        )
+        for rule in qml.list_decomps(qml.QROM):
+            _test_decomposition_rule(op, rule)
+
```

Comment:
I'd like to see more testing. Specifically:
* A test for `clean=False`
* More target wires and longer bitstrings
* Zero work wires, more than 1 work wire
* Zero control wires